analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java

```diff
@@ -159,7 +159,7 @@ public void setUp()
 		MessageDestination anonMessageDestinationEntity = MessageDestination
 				.createInstance(PublicKeyUtil.generateKeyPair().getPublic());
 		Set<Goal> goals = new HashSet<>(Arrays.asList(gamblingGoal, gamingGoal, socialGoal, shoppingGoal));
-		deviceAnonEntity = DeviceAnonymized.createInstance(0, OperatingSystem.IOS, "Unknown");
+		deviceAnonEntity = DeviceAnonymized.createInstance(0, OperatingSystem.IOS, "Unknown", Optional.empty());
```

Comment:
Only one out of the 11 calls to ``DeviceAnonymized.createInstance`` passes a nonempty value for the Firebase ID, so please add an overload that does not have the Firebase ID and make that pass the empty value to the current implementation.

Reply:
All those 10 calls that do not pass it are from test code. Adding a new method for test code only seems strange.

Reply:
That was my first thought, but there is one from non-test code. That made me switch and propose to add an overload.

Reply:
Nor can I. I was mistaken.